Python/emscripten_signal.c

```diff
@@ -38,19 +38,17 @@ _Py_CheckEmscriptenSignals(void)
     }
 }
 
-
 #define PY_EMSCRIPTEN_SIGNAL_INTERVAL 50
 static int emscripten_signal_clock = PY_EMSCRIPTEN_SIGNAL_INTERVAL;
 
 void
 _Py_CheckEmscriptenSignalsPeriodically(void)
 {
-    if (!Py_EMSCRIPTEN_SIGNAL_HANDLING) {
-        return;
-    }
-    emscripten_signal_clock--;
     if (emscripten_signal_clock == 0) {
```

Comment:
I still think we should keep the guard:
```suggestion
    if (!Py_EMSCRIPTEN_SIGNAL_HANDLING) {
        return;
    }
    if (emscripten_signal_clock == 0) {
```

Reply:
If `emscripten_signal_clock == 0 && Py_EMSCRIPTEN_SIGNAL_HANDLING == false` then this code will get stuck in a loop, where `RESUME_CHECK` calls `_Py_CheckEmscriptenSignalsPeriodically` because `emscripten_signal_clock == 0` but `_Py_CheckEmscriptenSignalsPeriodically` doesn't set `emscripten_signal_clock` to non-zero because `Py_EMSCRIPTEN_SIGNAL_HANDLING` is false.

We need to set `emscripten_signal_clock` to non-zero every time `_Py_CheckEmscriptenSignalsPeriodically` is called.

I think we have missed an update to `emscripten_signal_clock` though.
We need `emscripten_signal_clock--` in RESUME_CHECK to ensure that we actually check for signals.

Reply:
Ah yes I see what you mean.